deployment/environment/memory/environment.go

```diff
@@ -80,10 +81,11 @@ func generateMemoryChain(t *testing.T, inputs map[uint64]EVMChain) map[uint64]de
 				}
 				for {
 					backend.Commit()
-					receipt, err := backend.TransactionReceipt(context.Background(), tx.Hash())
-					if err != nil {
-						t.Log("failed to get receipt", "chain", chainInfo.ChainName, err)
-						continue
+					ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+					defer cancel()
+					receipt, err := bind.WaitMined(ctx, backend, tx)
```

Comment:
The linter normally complains about defers from for loops since they can be error prone. An alternative form is to wrap it up in a func:
```suggestion
					receipt, err := func() (receipt, error) {
						ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
						defer cancel()
						return bind.WaitMined(ctx, backend, tx)
					}
```